src/kubernetes.cc

```diff
@@ -792,30 +792,39 @@ void KubernetesReader::WatchMaster(
   http::client::request request(endpoint);
   request << boost::network::header(
       "Authorization", "Bearer " + KubernetesApiToken());
-  const bool verbose = config_.VerboseLogging();
-  if (verbose) {
-    LOG(INFO) << "WatchMaster(" << name << "): Contacting " << endpoint;
-  }
   try {
+    const bool verbose = config_.VerboseLogging();
     if (verbose) {
-      LOG(INFO) << "Locking completion mutex";
+      LOG(INFO) << "WatchMaster(" << name << "): Will retry "
+                << config_.KubernetesUpdaterWatchRetries() << " times";
     }
-    // A notification for watch completion.
-    std::mutex completion_mutex;
-    std::unique_lock<std::mutex> watch_completion(completion_mutex);
-    Watcher watcher(
-      endpoint,
-      [=](json::value raw_watch) {
-        WatchEventCallback(callback, name, std::move(raw_watch));
-      },
-      std::move(watch_completion), verbose);
-    http::client::response response = client.get(request, std::ref(watcher));
-    if (verbose) {
-      LOG(INFO) << "Waiting for completion";
+    for (int i = 0; i < config_.KubernetesUpdaterWatchRetries(); i++) {
+      if (verbose) {
+        LOG(INFO) << "WatchMaster(" << name << "): Contacting " << endpoint
+                  << " attempt #" << i;
```

Comment:
Nit: should we do `i + 1` to make it more human-friendly? If so, let's do it below as well.

Reply:
See my comments for an alternate proposal.

Reply:
Done.